app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -307,7 +307,7 @@ def populate_hardware_hash_with_disks(hardware_disks_array, instance)
         data_disks.each do |disk|
           disk_size      = disk.respond_to?(:disk_size_gb) ? disk.disk_size_gb * 1.gigabyte : 0
           disk_name      = disk.name
-          disk_location  = disk.try(:vhd).try(:uri)
+          disk_location  = disk.try(:vhd).try(:uri) || disk.try(:managed_disk).try(:id)
```

Comment:
`||` will get naughty if the first method exists but returns an empty string... 
We had that problem 3 times in amazon.

But I guess the azure sdk does a better job here :)